File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    public String solution(String s, String skip, int index) {
        StringBuilder sb = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            int cnt = 0;
            
            while (cnt < index) {
                c = (c == 'z') ? 'a' : (char)(c + 1);
                if (skip.contains(String.valueOf(c))) continue;
                cnt++;
            }
            
            sb.append(c);
        }
        
        return sb.toString();
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.kt
class Solution {
    fun solution(s: String, skip: String, index: Int): String {
        val alphabet = ('a' .. 'z')
            .filterNot { it in skip }
        val size = alphabet.size
        val map = alphabet.withIndex().associate { it.value to it.index }
        
        return s.map { c ->
            alphabet[(map[c]!! + index) % size]
        }.joinToString("")
    }
}